src/zenml/models/pipeline_models.py

```diff
@@ -121,6 +121,7 @@ class ArtifactModel(DomainModel):
 
     name: str  # Name of the output in the parent step
 
+    artifact_store_id: Optional[UUID]
     parent_step_id: UUID
```

Comment:
Tbh it feels a little weird to me that our `ArtifactModel` stores the parent step id and producer step id.
Querying cached artifacts from different runs returns different models even though they refer to the same underlying artifact.

But I wouldn't change any of that for now and just merge it, there will probably be more updates to all these models and schemas that we'll figure out along the way. We'll just have to make sure to take a look at the final updates once TFX is removed and we're actually going to merge the removal branch.

Reply:
I agree, let's revisit the models once TFX is removed

Reply:
See ENG-1500